### DISCUSSION TITLE
ch3/ch3-01

### DISCUSSION THREAD

**Discussion by giscus** [gopl-zh/gopl-zh.github.com#218] [General]
# ch3/ch3-01

<The Go Programming Language>中文版

https://gopl-zh.github.io/ch3/ch3-01.html

**Comment by pf-w** [comment]
### 位清空
z = x &^ y
如: x=14；y=6；
14 &^ 6
y的二进制表示为00000110，y的第二位和第三位为1，表示对x的第二位和第三位进行清空，
即：**对x的哪些位进行清空，就将y的哪些位设置为1**

&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;0 0 0 0 1 1 1 0
&^&nbsp;&nbsp;0 0 0 0 0 1 1 0
—----------------
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;0 0 0 0 1 0 0 0

**Comment by a-firefly** [comment]
x &^ y = x & (^y)

**Comment by mefly2012** [comment]
其中有符号整数采用2的补码形式表示，也就是最高bit位用来表示符号位，一个n-bit的有符号数的值域是从-2n-1到2n-1-1    这里应该写成次方的形式，-2ⁿ-1到2ⁿ-1-1（2的n-1次幂的值再-1）

**Comment by Apahno** [comment]
位清空
